src/components/TableComponent.tsx

```diff
@@ -0,0 +1,78 @@
+import { useParams } from 'react-router-dom';
+import { Person } from '../types';
+import { ErrorComponent } from './ErrorComponent';
+import { PersonLink } from './PersonLink';
+
+interface TableComponentProps {
+  people: Person[];
+  isLoadingError: boolean;
+}
+
+export function TableComponent({
+  people,
+  isLoadingError,
+}: TableComponentProps) {
+  const { slug } = useParams();
+
+  return (
+    <>
+      <ErrorComponent
+        isLoadingError={isLoadingError}
+        hasNoPeople={people.length === 0}
+      />
+      <table
+        data-cy="peopleTable"
+        className="table is-striped is-hoverable is-narrow is-fullwidth"
+      >
+        <thead>
+          <tr>
+            <th>Name</th>
+            <th>Sex</th>
+            <th>Born</th>
+            <th>Died</th>
+            <th>Mother</th>
+            <th>Father</th>
+          </tr>
+        </thead>
+
+        <tbody>
+          {people.map(person => {
+            const mother = people.find(p => p.name === person.motherName);
+
+            const father = people.find(p => p.name === person.fatherName);
```

Comment:
The logic for finding the mother and father assumes that names are unique, which might not always be the case. Consider using a unique identifier, such as an ID, if available, to find the correct parent.